src/avatars.js

```diff
@@ -1,13 +1,19 @@
 "use strict";
 
-import getAvatarURL from "gh-avatar";
-import axios from "axios";
+import got from "got";
 import sharp from "sharp";
+import createCache from "keyv-fs-sync";
 
-export const fetchEmbedableAvatar = async (username, token, {avatarSize}) => {
-  const url = await getAvatarURL(username, token);
-  const httpResult = await axios.get(url, {responseType: "arraybuffer"});
-  const resizedImage = await sharp(Buffer.from(httpResult.data))
+export const fetchEmbedableAvatar = async (
+  username,
+  token,
+  {avatarSize, cacheDir}
+) => {
+  const url = `https://github.com/${username.toLowerCase()}.png`;
```

Comment:
I'm noting that this code only works for GitHub identities, so it will break with the new scores format (and inclusion of SC and Discourse identities), yes?

Also: this is actually a reason that the scores format with address parts is better than just including aliases, as the address parts for the Discourse address include the server url, while the alias wouldn't. So with the address parts, we could infer the right avatar url for Discourse users.

In the longer run, I'd like to see us standardize more and more on SourceCred identities rather than adding custom avatar logic for every plugin that gets added.

Reply:
> I'm noting that this code only works for GitHub identities, so it will break with the new scores format (and inclusion of SC and Discourse identities), yes?

Indeed, see #32, #30, #37 

> Also: this is actually a reason that the scores format with address parts is better than just including aliases, as the address parts for the Discourse address include the server url, while the alias wouldn't. So with the address parts, we could infer the right avatar url for Discourse users.

Interesting point about addresses including the discourse domain, however I feel like that is as much a lucky side-effect of parsing an opaque identifier which the plugin developer is free to specify as it is a solution :sweat_smile:

> In the longer run, I'd like to see us standardize more and more on SourceCred identities rather than adding custom avatar logic for every plugin that gets added.

Agree with this. The process of merging those identities is the ideal point where we can decide where to source profile data from like avatars. Probably the same applies to the `<a href>` to add to each avatar in the output SVG. Some kind of "profile page" in the sourcecred explorer with some overview of their contributions and underlying identities would be very nice.

All of which I would consider outside of this particular PRs scope though.
In terms of future-proofing the image caching for other implementations. So long as it's based on HTTP requests with proper caching instructions in the response headers, this will be usable. The expected result is `cache/widgets/<hex encoded URLs>` with response data. It will do standards compliant requests, where a 304 response would be a cache hit. This should work regardless of whether you're hitting GitHub CDNs, self-hosted Discourse instances or a SourceCred profile page avatar.